lib/Controller/DocumentController.php

```diff
@@ -116,6 +116,13 @@ public function index($fileId) {
 				'instanceId' => $this->settings->getSystemValue('instanceid'),
 			];
 
+			// Update the current file to be accessible with system public shared key
+			$owner = $item->getOwner()->getUID();
+			$absPath = '/' . $owner . '/' .  $item->getInternalPath();
+			$accessList = \OC::$server->getEncryptionFilesHelper()->getAccessList($absPath);
+			$accessList['public'] = true;
+			\OC::$server->getEncryptionManager()->getEncryptionModule()->update($absPath, $owner, $accessList);
```

Comment:
In general your approach works. I see only one problem. With `\OC::$server->getEncryptionManager()->getEncryptionModule()` you always get the default encryption module, so in a scenario where multiple encryption modules exists it could happen that you use the wrong module.

I confess that this is a theoretical problem because as of now only one encryption module exists. But it could become a problem in the future. Unfortunately I don't have a easy suggestion how to do it different with the current API.